src/pal/src/file/file.cpp

```diff
@@ -999,6 +999,12 @@ CreateFileW(
     }
     
     name = namePathString.OpenStringBuffer(length);
+    if (NULL == name)
+    {
+        palError = ERROR_NOT_ENOUGH_MEMORY;
```

Comment:
This function does SetLastError before the "done" label. Could you please move SetLastError to be after "done:"?